designs/2022-unused-disable-directive-flexible-config/README.md

```diff
@@ -0,0 +1,228 @@
+
+1. Running linting with `eslint --report-unused-disable-directives` (errors, but requires providing the CLI option whenever running ESLint)
+2. Adding `reportUnusedDisableDirectives: true` to the `.eslintrc.js` config file (warns, and thus easily ignored)
+3. Enabling the third-party rule [eslint-comments/no-unused-disable](https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/no-unused-disable.html) (errors, but requires a third-party plugin)
+
+These existing solutions have various downsides:
+
+1. Lack of discoverability.
+2. Requires manual configuration to enable and thus not commonly used.
+3. Inability to treat unused disable directives as errors in shareable configs using the first-party options.
+4. Inconsistency in behavior between the CLI option and config file option of the same name which is unintuitive and causes confusion.
+
+### New Design
+
+We propose a new "explicit severity" design where the user can set the severity level of unused disable directive reporting using either the CLI option or the config file option. As a result, the user should always be able to achieve the desired effect (`off` / `warn` / `error`), regardless of where they set the option from (CLI option, config file option, or even from a shareable config).
+
+We will also change the default behavior to warn on unused disable directives so that all users (unless they opt-out) will benefit from the detection (and automatic removal with `--fix`) of unused disable directives.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+How the new "explicit severity" design works:
+
+- The config and CLI options are both changed to also accept standard severity level string values: `off`, `warn`, `error` (or a the corresponding number for each level)
+- If both options are present, the CLI option takes precedence over (overrides) the config option
+- If only one of the options is present, the value of the present option is used
+- If neither option is present, the default value (`warn`) is used
+- Note: There's only one underlying setting, and the config and CLI options are just two different ways of controlling it
+
+Boolean values will continue to be allowed with no change in behavior, as summarized below.
+
+Allowed values for the config file option:
+
+- `reportUnusedDisableDirectives: true` - same behavior as today (`warn`)
+- `reportUnusedDisableDirectives: false` - same behavior as today (`off`)
+- `reportUnusedDisableDirectives: "error"` - new value
+- `reportUnusedDisableDirectives: "warn"` - new value
+- `reportUnusedDisableDirectives: "off"` - new value
+
+Allowed values for the CLI option:
```

Comment:
It looks like we are missing the updated `--report-unused-disable-directives-severity` option here.

Reply:
Updated.